src/avro/src/decode.rs

```diff
@@ -8,6 +8,7 @@ use std::fs::File;
 use std::io::{self, Cursor, Read, Seek, SeekFrom};
 
 use chrono::{NaiveDate, NaiveDateTime};
+use flate2::read::MultiGzDecoder;
```

Comment:
Sorry you did the extra work in this file, but I don't think we should support gzipped Avro files, because Avro files already have their own internal compression scheme, which we already support automatically, as of https://github.com/MaterializeInc/materialize/pull/3815 .

Reply:
The reason this needs to be here is because the `read_file_task` expects an `dyn AvroRead` regardless of the file being read https://github.com/MaterializeInc/materialize/blob/main/src/dataflow/src/source/file.rs#L310